x-pack/plugins/file_upload/public/components/geojson_upload_form/index_name_form.tsx

```diff
@@ -5,23 +5,46 @@
  * 2.0.
  */
 
+import _ from 'lodash';
 import React, { ChangeEvent, Component } from 'react';
 import { EuiFormRow, EuiFieldText, EuiCallOut, EuiSpacer } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
-import { validateIndexName } from '../../util/indexing_service';
+import { validateIndexName } from '../../validate_index_name';
 
 export interface Props {
   indexName: string;
   indexNameError?: string;
   onIndexNameChange: (name: string, error?: string) => void;
+  onIndexNameValidationStart: () => void;
+  onIndexNameValidationEnd: () => void;
 }
 
 export class IndexNameForm extends Component<Props> {
-  _onIndexNameChange = async (event: ChangeEvent<HTMLInputElement>) => {
+  private _isMounted = false;
+
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  _onIndexNameChange = (event: ChangeEvent<HTMLInputElement>) => {
     const indexName = event.target.value;
+    this.props.onIndexNameChange(indexName);
```

Comment:
Do we need this call to update index name when it gets called again very shortly on line 46?

Reply:
Yes, the call to onIndexNameChange on line 35 updates indexName state as users type. If you did not update immediately and instead waited until the async task completed, then the newly typed text would not show up since indexName state would not change.